src/components/cards/remote-cards/RemoteCardProvider.tsx

```diff
@@ -0,0 +1,137 @@
+import { noop, pick } from 'lodash';
+import React, { PropsWithChildren, createContext, useState } from 'react';
+import { Card } from '@/graphql/__generated__/arc';
+import Routes from '@/navigation/routesNames';
+import { IS_TEST } from '@/env';
+import {
+  CardCollectionResult,
+  useCardCollectionQuery,
+} from '@/resources/cards/cardCollectionQuery';
+import * as ls from '@/storage';
+import { REMOTE_CARDS, useExperimentalFlag } from '@/config';
+import { useRemoteConfig } from '@/model/remoteConfig';
+
+const TRIMMED_CARD_KEYS = [
+  'cardKey',
+  'dismissable',
+  'placement',
+  'index',
+  'backgroundColor',
+  'accentColor',
+  'padding',
+  'imageIcon',
+  'imageRadius',
+  'title',
+  'titleColor',
+  'subtitle',
+  'subtitleColor',
+  'primaryButton',
+] as const;
+
+type RoutesWithIndex = typeof Routes & { [key: string]: string };
+
+export type TrimmedCard = Pick<Card, typeof TRIMMED_CARD_KEYS[number]> & {
+  sys: Pick<Card['sys'], 'id'>;
+  imageCollection: {
+    items: {
+      url: string;
+    }[];
+  };
+};
+
+type CardProviderProps = {
+  initialState?: Record<keyof TrimmedCard['cardKey'], TrimmedCard>;
+};
+
+type CardContextProps = {
+  cards: Record<string, TrimmedCard>;
+  setCards: React.Dispatch<React.SetStateAction<Record<string, TrimmedCard>>>;
+  dismissCard: (cardId: string) => void;
+  getCardsForPlacement: (placement: string) => TrimmedCard[];
+};
+
+export const RemoteCardContext = createContext<CardContextProps>({
+  cards: {},
+  setCards: noop,
+  dismissCard: noop,
+  getCardsForPlacement: () => [],
+});
+
+export const RemoteCardProvider: React.FC<
+  PropsWithChildren<CardProviderProps>
+> = ({ children }) => {
+  const config = useRemoteConfig();
+
+  const [cards, setCards] = useState<Record<string, TrimmedCard>>({});
+  const enabled =
+    useExperimentalFlag(REMOTE_CARDS) || config.remote_cards_enabled;
+
+  useCardCollectionQuery(
+    {},
+    {
+      enabled: enabled && !IS_TEST,
+      refetchInterval: 5_000,
```

Comment:
```suggestion
      refetchInterval: 60_000,
```
after development testing

Reply:
agree